src/main/java/com/fasterxml/jackson/databind/ext/OptionalHandlerFactory.java

```diff
@@ -54,7 +54,7 @@ public class OptionalHandlerFactory implements java.io.Serializable
         try {
             node = org.w3c.dom.Node.class;
             doc = org.w3c.dom.Document.class;
-        } catch (Throwable e) {
+        } catch (NoClassDefFoundError | Exception e) {
```

Comment:
Let's leave exception catching in this particular class as it was; we don't have tests to verify that loading would not fail via `Error`s. Even if sort of unlikely, I think `LinkageError` etc might occur.